gui/wxpython/main_window/notebook.py

```diff
@@ -0,0 +1,146 @@
+"""
+@package main_window.notebook
+
+@brief Custom AuiNotebook class and class for undocked AuiNotebook frame
+
+Classes:
+ - notebook::MapPageFrame
+ - notebook::MapNotebook
+
+(C) 2022 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author Linda Kladivova <lindakladivova gmail.com>
+@author Anna Petrasova <kratochanna gmail.com>
+"""
+
+import os
+
+import wx
+import wx.lib.agw.aui as aui
+
+from core import globalvar
+from gui_core.wrap import SimpleTabArt
+
+
+class MapPageFrame(wx.Frame):
+    """Frame for independent map display window."""
+
+    def __init__(self, parent, mapdisplay, size, pos, title):
+        wx.Frame.__init__(self, parent=parent, size=size, pos=pos, title=title)
+        self.mapdisplay = mapdisplay
+        self.sizer = wx.BoxSizer(wx.HORIZONTAL)
+        self.SetSizerAndFit(self.sizer)
```

Comment:
```suggestion
        self.SetSizer(self.sizer)
        self.Centre()
```

This solves the problem on Mac, specifically `self.SetSizerAndFit` was the culprit.
`self.Centre()` was needed as the detached panel was placed partly below the macOS main menu.

Reply:
Nicklas, thank you for testing it!
Unfortunately, on my Ubuntu the result does not look good after that change:
![Screenshot from 2023-02-13 17-37-37](https://user-images.githubusercontent.com/49241681/218518610-ccac4023-e087-4b69-a287-a635ca7dec59.png)
The content has been put a bit lower and a status bar is not visible.

So we will probably need to apply different solutions based on OS. It would be nice if someone tested it also on Windows. @hellik could I ask you? We would specifically need to verify that undocking and docking using the new Undock/Dock button works fine here.

Reply:
I think I had the same problem. But I was happy to see THAT much ! :-) , a manual window resize and everything good.
I'll see if I can come up with something regarding this detail.

Reply:
I have tried everything possible, that I could think of, but was unable to make this work as it should (so far). Only resizing "manually", even with as little as a pixel size change, would re-render the status bar and all content correctly.

Reply:
Not sure what you tried, but right now the mapdisplay is added to be managed by the sizer after the frame is created (see the UndockMapDisplay method), so it is not completely surprising this doesn't quite work. I suggest to move the `frame.sizer.Add(...)` call into the MapPageFrame.

Reply:
You are right @petrasovaa .. it is not so easy because we have the reparent method here. I have tried several options and the only thing I could think of is putting self.mapdisplay.Reparent(self) directly inside MapPageFrame __init__. SetSizer needs to be called after reparenting otherwise it does not work.

Now it works for me as expected. Please @nilason, could you test it one more time?

Reply:
With [e2f1db8](https://github.com/OSGeo/grass/pull/2667/commits/e2f1db8baa65a59cbca22a2f48ad5ac9a39f0414) it works for me too!

I still need a `frame.Centre(wx.BOTH)` for correct positioning though, not only will the undocked window be placed partially under macOS menu:

<img width="567" alt="Screenshot" src="https://user-images.githubusercontent.com/14186207/220100106-74bd3965-4946-4d83-b0f9-235ca9b17ad5.png">

but with multiple displays, the undocked window will always open in primary display (even if single-window is in secondary. With centre command the undocked window opens centred in the same display.

Reply:
I'm sorry, I'm not following this fully, but centering is best done on the parent than on the screen. Center on screen does not work well on extremely large displays like [this one](https://doi.org/10.3390/ijgi4020942):

![Screenshot from 2023-02-20 09-43-25](https://user-images.githubusercontent.com/5449060/220136999-2030cedd-4c3d-4ae5-8f9a-4c6a0892f2c0.png)